app/services/report_importer.rb

```diff
@@ -5,12 +5,13 @@ class ReportImporter
   attr_reader :report, :report_scanners
 
   # When writing your own Report importer, provide feature(s) of authorized Smart Proxies
+  # via ReportImporter.register_smart_proxy_feature method. Do not override this method!
   def self.authorized_smart_proxy_features
     @authorized_smart_proxy_features ||= []
   end
 
   def self.register_smart_proxy_feature(feature)
-    @authorized_smart_proxy_features = (authorized_smart_proxy_features + [feature]).uniq
+    @authorized_smart_proxy_features = (authorized_smart_proxy_features + [feature.freeze]).uniq
```

Comment:
didn't you want to freeze the array instead? You're creating new one here, so it would be fine and it would not allow touching the array anywhere else, then through this method. But freezing the strings sounds weird, are we worried someone is changing the actual strings?